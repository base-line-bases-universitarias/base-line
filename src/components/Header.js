import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => {
	return (
		<div className='header'>
			<nav className='header__nav'>
				<ul className='header__nav__list'>
					<li className='option'>
						<NavLink
							activeClassName='active'
							className='option__link'
							to='/'
							exact
						>
							Inicio
						</NavLink>
					</li>
					<li className='option'>
						<NavLink
							activeClassName='active'
							className='option__link'
							to='/contactenos'
							exact
						>
							Contactenos
						</NavLink>
					</li>
					<li className='option'>
						<NavLink
							activeClassName='active'
							className='option__link'
							to='/login'
							exact
						>
							Login
						</NavLink>
					</li>
				</ul>
			</nav>
		</div>
	);
};

export default Header;
