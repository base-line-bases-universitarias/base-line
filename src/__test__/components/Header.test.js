import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../components/Header';

describe('Esta es una swuit de pruebas', () => {
	test('Debe hacer render del componente Home', () => {
		const component = shallow(<Header />);
		expect(component.length).toEqual(1);
		expect(component).toMatchSnapshot();
	});
	test('Comprobar cantidad de link del menu principal', () => {
		const component = shallow(<Header />);
		const arr = component.find('.option__link').length;
		expect(arr).toEqual(3);
	});
	test('Comprobar texto de cada uno de los links del menu principal', () => {
		const component = shallow(<Header />);
		const links = ['Inicio', 'Contactenos', 'Login'];
		component.find('.option__link').forEach((item, index) => {
			expect(item.text()).toEqual(links[index]);
		});
	});
});
