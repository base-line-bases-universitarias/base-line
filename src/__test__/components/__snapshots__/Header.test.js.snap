// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Esta es una swuit de pruebas Debe hacer render del componente Home 1`] = `
<div
  className="header"
>
  <nav
    className="header__nav"
  >
    <ul
      className="header__nav__list"
    >
      <li
        className="option"
      >
        <NavLink
          activeClassName="active"
          className="option__link"
          exact={true}
          to="/"
        >
          Inicio
        </NavLink>
      </li>
      <li
        className="option"
      >
        <NavLink
          activeClassName="active"
          className="option__link"
          exact={true}
          to="/contactenos"
        >
          Contactenos
        </NavLink>
      </li>
      <li
        className="option"
      >
        <NavLink
          activeClassName="active"
          className="option__link"
          exact={true}
          to="/login"
        >
          Login
        </NavLink>
      </li>
    </ul>
  </nav>
</div>
`;
